src/runtime_src/core/pcie/driver/linux/xocl/subdev/ert_user.c

```diff
@@ -233,12 +235,98 @@ static ssize_t ert_intr_store(struct device *dev,
 }
 static DEVICE_ATTR_WO(ert_intr);
 
+static ssize_t mb_sleep_store(struct device *dev,
+	struct device_attribute *da, const char *buf, size_t count)
+{
+	struct platform_device *pdev = to_platform_device(dev);
+	u32 go_sleep;
+
+	if (kstrtou32(buf, 10, &go_sleep) == -EINVAL || go_sleep > 2) {
+		xocl_err(&to_platform_device(dev)->dev,
+			"usage: echo 0 or 1 > mb_sleep");
+		return -EINVAL;
+	}
+
+	if (go_sleep)
+		ert_user_gpio_cfg(pdev, MB_SLEEP);
+	else
+		ert_user_gpio_cfg(pdev, MB_WAKEUP);
```

Comment:
Do we need protect here? If this node is only for internal debug purpose, then without lock is okay.
Otherwise, we needs to deal with concurrently write.

Reply:
I added lock in ert_user_gpio_cfg to protect this section